EventFilter/HGCalRawToDigi/src/HGCalUnpacker.cc

```diff
@@ -60,11 +76,15 @@ void HGCalUnpacker::parseFEDData(unsigned fedId,
     std::cout << "fedId = " << fedId << ", captureblockIdx = " << captureblockIdx << ", cb_header = " << std::hex
               << std::setfill('0') << std::setw(16) << cb_header << std::dec << std::endl;
 
-    // parse ECON-Ds
-    for (uint16_t econdIdx = 0; econdIdx < HGCalMappingModuleIndexer::maxECONDperCB_; econdIdx++) {
+    // parse Capture Block body (ECON-Ds)
+    for (uint32_t econdIdx = 0; econdIdx < HGCalMappingModuleIndexer::maxECONDperCB_; econdIdx++) {
       auto econd_pkt_status = (cb_header >> (3 * econdIdx)) & 0b111;
       std::cout << "fedId = " << fedId << ", captureblockIdx = " << captureblockIdx << ", econdIdx = " << econdIdx
                 << ", econd_pkt_status = " << econd_pkt_status << std::endl;
+      if (econd_pkt_status != backend::ECONDPacketStatus::InactiveECOND) {
+        // always increment the global ECON-D index (unless inactive/unconnected)
+        globalECONDIdx++;
```

Comment:
@pfs Is this the correct way to increment the global counter?

Reply:
Yes this could work. My only points of reluctance are:
- like this we rely on the BE flags
- we don't yet have a inactive/masking mechanism in the global map
The second point is of course temporary.